ckanext-validation-schema-generator work

duttonw opened the pull request:
Have reset master to initial comment and bumped work into 'develop' branch so PR code review can occur.

mutantsan committed:
remove redundant folders

mutantsan committed:
webassets

mutantsan committed:
create a basic index page for schema generation

mutantsan committed:
style page

mutantsan committed:
write few auth tests, prepare test env

mutantsan committed:
remote test_plugin.py

mutantsan committed:
implement job and related staff

mutantsan committed:
change error status wording

mutantsan committed:
fix bug with json.loads empty schema

mutantsan committed:
remove i18n

mutantsan committed:
update actions

mutantsan committed:
write tests

mutantsan committed:
write readme

mutantsan committed:
remove whitespaces for generation table

ThrawnCA commented on ckanext/validation_schema_generator/logic/action.py:
```diff
@@ -0,0 +1,208 @@
+    :param id: resource ID
+    :type id: string
+    """
+    l.check_access('vsg_generate', context, data_dict)
```
Why have a custom access check at all? Is there any situation where it would have different permissions from `resource_update`?

ThrawnCA commented on ckanext/validation_schema_generator/logic/action.py:
```diff
@@ -0,0 +1,208 @@
+import json
+
+import ckan.plugins.toolkit as tk
+import ckan.logic as l
```
Why not use `tk`?

ThrawnCA commented on ckanext/validation_schema_generator/logic/action.py:
```diff
@@ -0,0 +1,208 @@
+import ckan.plugins.toolkit as tk
+import ckan.logic as l
+from ckan.common import config, asint
+from ckan.lib.jobs import enqueue as enqueue_job
```
`tk` supplies this.

ThrawnCA commented on ckanext/validation_schema_generator/logic/action.py:
```diff
@@ -0,0 +1,208 @@
+
+import ckan.plugins.toolkit as tk
+import ckan.logic as l
+from ckan.common import config, asint
```
`tk` supplies this.

ThrawnCA commented on ckanext/validation_schema_generator/logic/action.py:
```diff
@@ -0,0 +1,208 @@
+from ckan.common import config, asint
+from ckan.lib.jobs import enqueue as enqueue_job
+
+import ckanext.validation_schema_generator.jobs as jobs
```
Three of these statements could be combined.

ThrawnCA commented on ckanext/validation_schema_generator/logic/action.py:
```diff
@@ -0,0 +1,208 @@
+
+
+def _unapply_resource_schema(resource_id):
+    """Unapply resource schema actually means applying an empty one"""
```
Why have an unapply function at all?

ThrawnCA commented on ckanext/validation_schema_generator/plugin/__init__.py:
```diff
@@ -0,0 +1,58 @@
+from ckanext.validation_schema_generator.helpers import _get_helpers
+
+try:
+    tk.requires_ckan_version("2.9")
```
It is cleaner to use `check_ckan_version`, which just returns True/False, than to throw and catch an exception.

ThrawnCA commented on ckanext/validation_schema_generator/plugin/__init__.py:
```diff
@@ -0,0 +1,58 @@
+    # IConfigurer
+
+    def update_config(self, config_):
+        tk.add_template_directory(config_, '../templates')
```
It would be nicer to put this module one level higher so we don't need relative paths, and just put the Pylons and Flask specific features in a sub-module.

ThrawnCA commented on ckanext/validation_schema_generator/templates/base.html:
```diff
@@ -0,0 +1,8 @@
+{% ckan_extends %}
```
What would this extend? There is no core template with this name.

ThrawnCA commented on ckanext/validation_schema_generator/templates/base.html:
```diff
@@ -0,0 +1,8 @@
+{% block scripts %}
+    {{ super() }}
+
+    {% set type = 'asset' if h.ckan_version().split('.')[1] | int >= 9 else 'resource' %}
```
This would break if the CKAN major version were updated (eg CKAN 3). It would be better to incorporate a call to `toolkit.check_ckan_version`

ThrawnCA commented on ckanext/validation_schema_generator/tests/test_action.py:
```diff
@@ -0,0 +1,279 @@
+
+import pytest
+
+import ckan.logic as logic
```
It would be better to use `toolkit`

ThrawnCA commented on ckanext/validation_schema_generator/tests/test_action.py:
```diff
@@ -0,0 +1,279 @@
+        resource = helpers.call_action("resource_show", id=resource["id"])
+        assert resource[const.RES_SCHEMA_FIELD]
+
+    def test_apply_for_package(self, table_schema):
```
This test never calls the "vsg" functions on a package.

ThrawnCA commented on ckanext/validation_schema_generator/views/__init__.py:
```diff
@@ -0,0 +1,156 @@
+import json
```
Why is this file in a submodule, instead of just being `ckanext/validation_schema_generator/views.py`?

ThrawnCA commented on ckanext/validation_schema_generator/views/__init__.py:
```diff
@@ -0,0 +1,156 @@
+from flask import Blueprint
+from flask.views import MethodView
+
+import ckan.lib.helpers as h
```
Why not just use `tk.h`?

duttonw committed:
Get CICD working

*  Move plugin back to root to follow other plugins standard
*  Add helper method for mixin selection
*  Add ckanext-scheming.git@release-1.2.0 to dev-requirements

duttonw committed:
Merge pull request #2 from qld-gov-au/code_improvements

Get CICD working

duttonw committed:
lock factory-boy for python2 module, swap out singular test for triplet

duttonw committed:
remove sysadmin factory, it does not look like its in use

duttonw committed:
Merge pull request #3 from qld-gov-au/code_improvements

dev dependancies cleanup

ThrawnCA committed:
generate alternative validation function for CKAN 2.8

- The 'dict_only' validator doesn't exist on 2.8

ThrawnCA committed:
make Flake8 happy

- clean up whitespace
- drop unused imports
- drop duplicate test function
- add Flake8 config file so we can run it manually

ThrawnCA committed:
reduce duplicate GitHub Actions workflow runs

- Don't run on every pull request when we already run on 'push'. Only run on pull requests to master,
to ensure we test cross-repository pull requests.

ThrawnCA committed:
Merge pull request #4 from qld-gov-au/code_improvements

Fix for CKAN 2.8

ThrawnCA commented:
There are a bunch of items that seem less than ideal, but probably no outright blockers.
